torch_geometric/nn/pool/knn.py

```diff
@@ -216,3 +216,65 @@ def __init__(self, emb: Optional[Tensor] = None):
     def _create_index(self, channels: int):
         import faiss
         return faiss.IndexFlatIP(channels)
+
+
+class ApproxKNNIndex(KNNIndex):
```

Comment:
i made it a base class since most of the code between the L2 and MIPS approx KNNs is the same except choosing the metric and i wanted to avoid code duplication. if you still think i should remove it i can though, let me know